Source/Scene/GroundPolylinePrimitive.js

```diff
@@ -0,0 +1,856 @@
+         */
+        this.geometryInstances = options.geometryInstances;
+
+        var appearance = options.appearance;
+        if (!defined(appearance)) {
+            appearance = new PolylineMaterialAppearance();
+        }
+        /**
+         * The {@link Appearance} used to shade this primitive. Each geometry
+         * instance is shaded with the same appearance.  Some appearances, like
+         * {@link PolylineColorAppearance} allow giving each instance unique
+         * properties.
+         *
+         * @type Appearance
+         *
+         * @default undefined
+         */
+        this.appearance = appearance;
+
+        /**
+         * Determines if the primitive will be shown.  This affects all geometry
+         * instances in the primitive.
+         *
+         * @type {Boolean}
+         *
+         * @default true
+         */
+        this.show = defaultValue(options.show, true);
+
+        /**
+         * This property is for debugging only; it is not for production use nor is it optimized.
+         * <p>
+         * Draws the bounding sphere for each draw command in the primitive.
+         * </p>
+         *
+         * @type {Boolean}
+         *
+         * @default false
+         */
+        this.debugShowBoundingVolume = defaultValue(options.debugShowBoundingVolume, false);
+
+        // Shadow volume is shown by removing a discard in the shader, so this isn't toggleable.
+        this._debugShowShadowVolume = defaultValue(options.debugShowShadowVolume, false);
+
+        this._primitiveOptions = {
+            geometryInstances : undefined,
+            appearance : undefined,
+            vertexCacheOptimize : defaultValue(options.vertexCacheOptimize, false),
+            interleave : defaultValue(options.interleave, false),
+            releaseGeometryInstances : defaultValue(options.releaseGeometryInstances, true),
+            allowPicking : defaultValue(options.allowPicking, true),
+            asynchronous : defaultValue(options.asynchronous, true),
+            compressVertices : defaultValue(options.compressVertices, true),
+            _createShaderProgramFunction : undefined,
+            _createCommandsFunction : undefined,
+            _updateAndQueueCommandsFunction : undefined
+        };
+
+        // Used when inserting in an OrderedPrimitiveCollection
+        this._zIndex = undefined;
+
+        this._ready = false;
+        this._readyPromise = when.defer();
+
+        this._primitive = undefined;
+
+        this._sp = undefined;
+        this._spPick = undefined;
+        this._sp2D = undefined;
+        this._spPick2D = undefined;
+        this._spMorph = undefined;
+        this._spPickMorph = undefined;
+        this._renderState = RenderState.fromCache({
+            cull : {
+                enabled : true // prevent double-draw. Geometry is "inverted" (reversed winding order) so we're drawing backfaces.
+            },
+            blending : BlendingState.ALPHA_BLEND
```

Comment:
Shouldn't `depthMask` be `false`? Have you tried drawing anything hovering above a line or picking the depth of terrain behind a line but would hit the invisible volume?